platform/eureka/client.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eureka
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type application struct {
+	Name      string      `json:"name"`
+	Instances []*instance `json:"instance"`
+}
+
+type instance struct {
+	Hostname   string   `json:"hostName"`
```

Comment:
According to the docs, hostName is used during access (not sure if its AWS specific), while there is also a virtualHostName and secureVirtualHostName(). Question is, when I say http://reviews.foo.local/getReviews, is `reviews.foo.local` coming from hostName or virtualHostName ?
@cmluciano ?